### PR TITLE
sFlow bugfixes

### PR DESCRIPTION
**Pull request by GarrickHe**
* Fix: issue where invalid polling intervals were accepted
* Fix: sFlow collector name issue
* Fix: sflow.xml error due to missing ;
* Update: sonic-type XML to handle sFlow PTYPE
* Update: sflow.xml to use new PTYPE
* Fix: Error when adding collectors
* Fix: Add exception handling

### CODE REVIEW

File: src/CLI/clitree/cli-xml/sonic_types.xml [hunk 1]
```diff
@@ -514,28 +520,42 @@ limitations under the License.
     <PTYPE
         name="PTP_DELAY_MECHANISM"
         method="select"
-		pattern="e2e(E2E) p2p(P2P)"
+	pattern="e2e(E2E) p2p(P2P)"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/sonic_types.xml]: Fix indentation here and in below lines

File: src/CLI/clitree/cli-xml/sonic_types.xml [hunk 2]
```diff
@@ -514,28 +520,42 @@ limitations under the License.
...
         method="select"
-		pattern="transparent(transparent) boundary(boundary) disable(disable)"
+	pattern="transparent(transparent) boundary(boundary) disable(disable)"
+        help=""
+        />
+    <!--=======================================================-->
+    <PTYPE
+        name="SFLOW_POLL_INTERVAL"
+        method="unsignedInteger"
+        pattern="0..300"
```
Comment (jeffyin-dell) [inline src/CLI/clitree/cli-xml/sonic_types.xml:551]: You could do a regex here to limit it to 0,5-300

**Review by jeffyin-dell — CHANGES_REQUESTED** [review]

File: src/CLI/clitree/cli-xml/sonic_types.xml (hunk 2 above)
Comment (GarrickHe) [reply src/CLI/clitree/cli-xml/sonic_types.xml:551]: This is caught by the sFlow YANG which has better error message than the xml file.

File: src/CLI/clitree/cli-xml/sonic_types.xml (hunk 1 above)
Comment (GarrickHe) [reply src/CLI/clitree/cli-xml/sonic_types.xml]: Will fix.

**Review by jeffyin-dell — APPROVED** [review]

**Pull request merged**